Modules/_testcapi/numbers.c

```diff
@@ -1,7 +1,168 @@
 #include "parts.h"
 #include "util.h"
 
+
+static PyObject *
+number_check(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    return PyLong_FromLong(PyNumber_Check(obj));
+}
+
+#define BINARYFUNC(funcsuffix, methsuffix)                           \
+    static PyObject *                                                \
+    number_##methsuffix(PyObject *Py_UNUSED(module), PyObject *args) \
+    {                                                                \
+        PyObject *o1, *o2;                                           \
+                                                                     \
+        if (!PyArg_ParseTuple(args, "OO", &o1, &o2)) {               \
+            return NULL;                                             \
+        }                                                            \
+                                                                     \
+        NULLABLE(o1);                                                \
+        NULLABLE(o2);                                                \
+        return PyNumber_##funcsuffix(o1, o2);                        \
+    };
+
+BINARYFUNC(Add, add)
+BINARYFUNC(Subtract, subtract)
+BINARYFUNC(Multiply, multiply)
+BINARYFUNC(MatrixMultiply, matrixmultiply)
+BINARYFUNC(FloorDivide, floordivide)
+BINARYFUNC(TrueDivide, truedivide)
+BINARYFUNC(Remainder, remainder)
+BINARYFUNC(Divmod, divmod)
+
+#define TERNARYFUNC(funcsuffix, methsuffix)                          \
+    static PyObject *                                                \
+    number_##methsuffix(PyObject *Py_UNUSED(module), PyObject *args) \
+    {                                                                \
+        PyObject *o1, *o2, *o3;                                      \
+                                                                     \
+        if (!PyArg_ParseTuple(args, "OOO", &o1, &o2, &o3)) {         \
+            return NULL;                                             \
+        }                                                            \
+                                                                     \
+        NULLABLE(o1);                                                \
+        NULLABLE(o2);                                                \
```

Comment:
```suggestion
        NULLABLE(o2);                                                \
        NULLABLE(o3);                                                \
```
And add corresponding tests.

You can also make the third argument optional.

Reply:
> You can also make the third argument optional.

Yep, I did that.